[BUG] workspace/symbol падает с ScopeNotActiveException (scopedTarget.workspaceSymbolIndex) в 1.0.0-rc.2

**Версия**
1.0.0-rc.2 (Windows-сборка, Zulu JDK 25)

**Описание ошибки диагностики**
Запрос workspace/symbol стабильно возвращает -32603 Internal error. В логе сервера:

  [ERROR] [workspace-service-1] [org.eclipse.lsp4j.jsonrpc.RemoteEndpoint]: Internal error:
  org.springframework.beans.factory.support.ScopeNotActiveException:
  Error creating bean with name 'scopedTarget.workspaceSymbolIndex':
  Scope 'workspace' is not active for the current thread; consider defining a scoped proxy
  for this bean if you intend to refer to it from a singleton

  Похоже на тот же корень, что и в #4120 (completionItem/resolve): обработчик обращается к workspace-scoped бину (workspaceSymbolIndex) из потока без
  установленного workspace-контекста. #4120 закрыл путь completion через withFreshDocumentContext, но путь workspace/symbol (фича #4105), судя по
  всему, не покрыт.

  Шаги воспроизведения

  1. Открыть достаточно большую конфигурацию 1С (наблюдалось на ~9000+ BSL-файлов).
  2. Любой LSP-клиент отправляет workspace/symbol (у нас — в т.ч. на этапе прогрева).
  3. Сервер бросает исключение выше; клиент получает -32603, выдача — 0 символов.

  Ожидаемо: workspace/symbol возвращает ранжированный список символов (как в 0.28.x).

  Окружение: клиент — mcp-lsp-bridge (LSP по stdio). На 0.28.3 тот же запрос работает корректно (проверено A/B на той же конфигурации). С VS Code
  независимо не проверялось, но природа фикса #4120 (серверная установка контекста потока) указывает на независимость от клиента.

--- a/src/main/java/com/github/_1c_syntax/bsl/languageserver/types/index/WorkspaceSymbolIndex.java
+++ b/src/main/java/com/github/_1c_syntax/bsl/languageserver/types/index/WorkspaceSymbolIndex.java
@@ -29,7 +29,6 @@ import com.github._1c_syntax.bsl.languageserver.context.symbol.SourceDefinedSymb
 import com.github._1c_syntax.bsl.languageserver.context.symbol.Symbol;
 import com.github._1c_syntax.bsl.languageserver.context.symbol.VariableSymbol;
 import com.github._1c_syntax.bsl.languageserver.context.symbol.variable.VariableKind;
-import com.github._1c_syntax.bsl.languageserver.infrastructure.WorkspaceScope;
 import com.github._1c_syntax.bsl.mdo.MD;
 import com.github._1c_syntax.bsl.types.ScriptVariant;
 import org.apache.commons.collections4.trie.PatriciaTrie;
@@ -105,14 +104,21 @@ import java.util.concurrent.locks.ReentrantReadWriteLock;
  * Индекс наследует {@link AbstractDocumentLifecycleClearableIndex}: его {@code @EventListener}'ы
  * сбрасывают записи документа на изменение содержимого, освобождение данных, закрытие и удаление.
  * <p>
+ * Бин — обычный синглтон (НЕ {@code @WorkspaceScope}): {@code workspace/symbol} по протоколу LSP —
+ * кросс-воркспейсный запрос, поэтому единый индекс хранит символы документов всех рабочих областей
+ * и отдаёт их одним ранжированным списком. Записи ключуются по URI документа ({@link Entry#uri()}),
+ * а жизненный цикл держится на per-document событиях (в т.ч. {@code ServerContextDocumentRemovedEvent},
+ * который рассылается и при удалении рабочей области), так что закрытие/удаление области штатно
+ * вычищает её документы из индекса без отдельного per-workspace teardown'а. Индекс самодостаточен:
+ * не обращается ни к одному {@code @WorkspaceScope}-бину, поэтому workspace-контекст потока ему не нужен.
+ * <p>
  * Потокобезопасность: {@link PatriciaTrie} не потокобезопасен, поэтому доступ к нему защищён
  * {@link ReadWriteLock}. Запросы ({@link #search(String, CancelChecker)}) держат read-lock,
  * переиндексация и сброс ({@link #index(DocumentContext)}, {@link #clear(URI)}) — write-lock,
- * так что чтение и запись не гоняются. Записи {@link Entry} неизменяемы и безопасно покидают
- * блокировку в составе результата.
+ * так что чтение и запись не гоняются (в т.ч. при конкурентной индексации из populate-пулов разных
+ * рабочих областей). Записи {@link Entry} неизменяемы и безопасно покидают блокировку в составе результата.
  */
 @Component
-@WorkspaceScope
 public class WorkspaceSymbolIndex extends AbstractDocumentLifecycleClearableIndex {
 
   /**

--- a/src/test/java/com/github/_1c_syntax/bsl/languageserver/BSLWorkspaceServiceTest.java
+++ b/src/test/java/com/github/_1c_syntax/bsl/languageserver/BSLWorkspaceServiceTest.java
@@ -41,6 +41,7 @@ import org.eclipse.lsp4j.FileRename;
 import org.eclipse.lsp4j.RenameFilesParams;
 import org.eclipse.lsp4j.WorkspaceFolder;
 import org.eclipse.lsp4j.WorkspaceFoldersChangeEvent;
+import org.eclipse.lsp4j.WorkspaceSymbolParams;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
@@ -576,6 +577,47 @@ class BSLWorkspaceServiceTest {
     } finally {
       applicationContext.removeApplicationListener(listener);
     }
+  }
+
+  /**
+   * Регрессия на ScopeNotActiveException из issue #4123: запрос {@code workspace/symbol}
+   * выполняется на потоке {@code workspaceServiceExecutor}, а JSON-RPC диспетчер не устанавливает
+   * workspace-контекст вызывающего потока. Раньше {@code workspaceSymbolIndex} был {@code @WorkspaceScope}
+   * и без контекста не резолвился ({@code ScopeNotActiveException: Scope 'workspace' is not active}).
+   * После де-скопинга индекс — обычный синглтон, поэтому запрос обслуживается с любого потока без
+   * установленного workspace-контекста.
+   */
+  @Test
+  void symbol_worksFromThreadWithoutWorkspaceContext() throws Exception {
+    // given
+    var workspaceUri = Absolute.uri(tempDir.toUri());
+    var testFile = tempDir.resolve("symbol_index_module.bsl").toFile();
+    FileUtils.writeStringToFile(
+      testFile,
+      "Процедура ТестоваяПроцедураПоискаСимволов() КонецПроцедуры",
+      StandardCharsets.UTF_8
+    );
+    var uri = Absolute.uri(testFile.toURI());
+
+    // Наполняем индекс: rebuildDocument синхронно публикует DocumentContextContentChangedEvent.
+    WorkspaceContextHolder.run(workspaceUri, () -> {
+      var ctx = workspaceServerContext();
+      var documentContext = ctx.addDocument(uri);
+      ctx.rebuildDocument(documentContext);
+    });
+
+    // Воспроизводим прод-условие: вызывающий поток (JSON-RPC диспетчер) без workspace-контекста.
+    WorkspaceContextHolder.clear();
+
+    var params = new WorkspaceSymbolParams("ТестоваяПроцедураПоискаСимволов");
+
+    // when
+    var symbols = workspaceService.symbol(params).get().getRight();
+
+    // then
+    assertThat(symbols)
+      .as("workspace/symbol должен обслуживаться с потока без workspace-контекста (индекс — синглтон)")
+      .anyMatch(symbol -> symbol.getName().equals("ТестоваяПроцедураПоискаСимволов"));
   }
 
   @Test

